tests/Base/StringUtilityInterfaceTest.php

```diff
@@ -254,6 +268,43 @@ public function removeSuffixDataProvider()
 
 
 
+    public function substringDataProvider()
+    {
+        return [
+            'only unique words' => [
+                'string'            => 'a line with only unique words',
+                'after'             => 'with',
+                'before'            => 'unique',
+                'expectedResult'    => ' only '
+            ],
+            'duplicated words' => [
+                'string'            => 'a line with some duplicated words like line some or words',
+                'after'             => 'some',
+                'before'            => 'words',
+                'expectedResult'    => ' duplicated '
+            ],
+            'no after' => [
+                'string'            => 'no after available here',
+                'after'             => 'some',
+                'before'            => 'here',
+                'expectedResult'    => 'no after available '
+            ],
+            'no before' => [
+                'string'            => 'no before available here',
+                'after'             => 'before',
+                'before'            => 'some',
+                'expectedResult'    => ' available here'
+            ],
+            'no after and no before' => [
+                'string'            => 'no before and after available here',
+                'after'             => 'some',
+                'before'            => 'word',
+                'expectedResult'    => 'no before and after available here'
+            ]
```

Comment:
Das mache ich sehr gern. Ich denke das ich alle Aenderungen in einem Pull-Request packen werde.